combat-trainer.lic

```diff
@@ -2207,9 +2207,8 @@ class TrainerProcess
     return unless Time.now - UserVars.almanac_last_use >= 600
     return if left_hand && !game_state.currently_whirlwinding
     unless @almanac_skills.empty?
-      training_skill = @almanac_skills
-                       .select { |skill| DRSkill.getxp(skill) < 18 }
-                       .sort_by { |skill| DRSkill.getxp(skill) }.first
+      training_skills = @almanac_skills.select { |skill| DRSkill.getxp(skill) < 18 }
+      training_skill = game_state.sort_by_rate_then_rank(training_skills).first
```

Comment:
Original logic is being updated to sort by rank, too. As well as to make that decision consistent throughout combat-trainer by using game_state.sort_by_rate_then_rank. If the decision algoritm should change in the future, there's now one spot to do it.